other/word_search.py

```diff
@@ -0,0 +1,395 @@
+"""
+Creates a random wordsearch with eight different directions
+that are best described as compass locations.
+
+@ https://en.wikipedia.org/wiki/Word_search
+"""
+
+
+from random import choice, randint, shuffle
+
+# The words to display on the word search -
+# can be made dynamic by randonly selecting a certain number of
+# words from a predefined word file, while ensuring the character
+# count fits within the matrix size (n x m)
+WORDS = ["cat", "dog", "snake", "fish"]
+
+WIDTH = 10
+HEIGHT = 10
+
+
+class WordSearch:
+    """
+    >>> ws = WordSearch(WORDS, WIDTH, HEIGHT)
+    >>> ws.board  # doctest: +ELLIPSIS
+    [[None, ..., None], ..., [None, ..., None]]
+    >>> ws.generate_board()
+    """
+
+    def __init__(self, words: list[str], width: int, height: int) -> None:
+        self.words = words
+        self.width = width
+        self.height = height
+
+        # Board matrix holding each letter
+        self.board: list[list[str | None]] = [[None] * width for _ in range(height)]
+
+    def insert_north(self, word: str, rows: list[int], cols: list[int]) -> None:
+        """
+        >>> ws = WordSearch(WORDS, 3, 3)
+        >>> ws.insert_north("cat", [2], [2])
+        >>> ws.board  # doctest: +NORMALIZE_WHITESPACE
+        [[None, None, 't'],
+        [None, None, 'a'],
+        [None, None, 'c']]
+        >>> ws.insert_north("at", [0, 1, 2], [2, 1])
+        >>> ws.board  # doctest: +NORMALIZE_WHITESPACE
+        [[None, 't', 't'],
+        [None, 'a', 'a'],
+        [None, None, 'c']]
+        """
+        word_length = len(word)
+        # Attempt to insert the word into each row and when successful, exit
+        for row in rows:
+            # Check if there is space above the row to fit in the word
+            if word_length > row + 1:
+                continue
+
+            # Attempt to insert the word into each column
+            for col in cols:
+                # Only check to be made here is if there are existing letters
+                # above the column that will be overwritten
+                letters_above = [self.board[row - i][col] for i in range(word_length)]
+                if all(letter is None for letter in letters_above):
+                    # Successful, insert the word north
+                    for i in range(word_length):
+                        self.board[row - i][col] = word[i]
+                    return
+
+    def insert_north_east(self, word: str, rows: list[int], cols: list[int]) -> None:
```

Comment:
```suggestion
    def insert_northeast(self, word: str, rows: list[int], cols: list[int]) -> None:
```
Nitpick: `north_east` -> `northeast`